src/watcher/store.ts

```diff
@@ -116,16 +121,17 @@ export class AppointmentStore extends StartStopService implements IAppointmentSt
             // remove the appointment from the id index
             delete this.appointmentsById[appointmentId];
 
-            // remove the appointment from the state locator index
-            const currentAppointment = this.appointmentsByStateLocator[appointmentById.getStateLocator()];
-            // if it has the same id
-            if (currentAppointment.id === appointmentId) {
-                delete this.appointmentsByStateLocator[appointmentById.getStateLocator()];
-            }
-
-            // and remove from remote storage
-            await this.db.del(appointmentId);
-
+            await this.lockManager.withLock(appointmentById.getStateLocator(), async () => {
```

Comment:
At the moment this isnt strictly necessary since the lock contents is blocking code. I'm tempted to say the lock should remain here anyway, as it might be easy for someone to put an await in here and forget to apply the lock. What are your thoughts?

Reply:
Left the lock, as discussed.